controllers/clusterresources/nodereload_controller.go

```diff
@@ -60,151 +68,188 @@ func (r *NodeReloadReconciler) Reconcile(ctx context.Context, req ctrl.Request)
+			"currentOperationStatus", nodeReloadStatus,
 		)
-		r.EventRecorder.Eventf(
-			nrs, models.Warning, models.PatchFailed,
-			"Resource status patch is failed. Reason: %v",
-			err,
+		r.EventRecorder.Eventf(nrs, models.Warning, models.FetchFailed,
+			"Failed to patch current operation status. Reason: %w", err,
 		)
-		return models.ReconcileRequeue, nil
+
+		return reconcile.Result{}, err
 	}
 
-	if nrs.Status.CurrentOperationStatus.Status != "COMPLETED" {
+	if nodeReloadStatus.Status != nodeReloadOperationStatusCompleted {
 		l.Info("Node Reload operation is not completed yet, please wait a few minutes",
 			"nodeID", nrs.Status.NodeInProgress,
 			"status", nrs.Status,
 		)
+
 		return models.ReconcileRequeue, nil
 	}
 
-	nrs.Status.NodeInProgress.ID = ""
+	l.Info("The node has been successfully reloaded",
+		"Node ID", nrs.Status.NodeInProgress.ID,
+	)
+	r.EventRecorder.Eventf(nrs, models.Normal, models.UpdatedEvent,
+		"Node %s has been successfully reloaded", nrs.Status.NodeInProgress.ID,
+	)
+
+	patch = nrs.NewPatch()
+
+	nrs.Status.CompletedNodes = append(nrs.Status.CompletedNodes, nrs.Status.NodeInProgress)
+	nrs.Status.CurrentOperationStatus, nrs.Status.NodeInProgress = nil, nil
+
 	err = r.Status().Patch(ctx, nrs, patch)
 	if err != nil {
-		l.Error(err,
-			"Cannot patch Node Reload status",
-			"nodeID", nrs.Status.NodeInProgress,
-		)
-		r.EventRecorder.Eventf(
-			nrs, models.Warning, models.PatchFailed,
-			"Resource status patch is failed. Reason: %v",
-			err,
+		l.Error(err, "Failed to patch completed nodes")
+		r.EventRecorder.Eventf(nrs, models.Warning, models.PatchFailed,
+			"Failed to patch completed nodes. Reason: %w", err,
 		)
-		return models.ReconcileRequeue, nil
+
+		return reconcile.Result{}, err
+	}
+
+	return models.ImmediatelyRequeue, nil
+}
+
+func (r *NodeReloadReconciler) handleNodeNotFound(ctx context.Context, node *v1beta1.Node, nrs *v1beta1.NodeReload) (reconcile.Result, error) {
+	l := log.FromContext(ctx)
+
+	patch := nrs.NewPatch()
+
+	if nrs.Status.NodeInProgress == nil && len(nrs.Status.PendingNodes) > 0 {
+		nrs.Status.PendingNodes = nrs.Status.PendingNodes[1:]
 	}
 
-	nrs.Spec.Nodes = nrs.Spec.Nodes[:len(nrs.Spec.Nodes)-1]
-	err = r.Patch(ctx, nrs, patch)
+	nrs.Status.FailedNodes = append(nrs.Status.FailedNodes, node)
+	nrs.Status.CurrentOperationStatus, nrs.Status.NodeInProgress = nil, nil
+
+	err := r.Status().Patch(ctx, nrs, patch)
 	if err != nil {
-		l.Error(err, "Cannot patch Node Reload cluster",
-			"spec", nrs.Spec,
-		)
-		r.EventRecorder.Eventf(
-			nrs, models.Warning, models.PatchFailed,
-			"Resource patch is failed. Reason: %v",
-			err,
+		l.Error(err, "Cannot patch failed node")
+		r.EventRecorder.Event(nrs, models.Warning, models.PatchFailed,
+			"Cannot patch failed node",
 		)
-		return models.ReconcileRequeue, nil
+
+		return reconcile.Result{}, err
 	}
 
-	return models.ExitReconcile, nil
+	l.Error(err, "Node is not found on Instaclustr",
```

Comment:
Node is not found on the Instaclustr side, something like that

Reply:
Changed to `Node is not found on the Instaclustr side`